workspace.go

```diff
@@ -349,3 +350,18 @@ func (ws *WorkspaceConfig) AddProviders(providers []Provider) {
 	ws.Providers = providerConfigs
 	ws.Terraform.RequiredProviders = versions
 }
+
+// PlanWorkspaceDeletion parses a plan to look for whether the delete action is associated with any target resource
+func PlanForDeletion(plan *tfjson.Plan, targetType string) bool {
```

Comment:
Remember that types are part of the fn signature, so something like `WillDestroy(plan *tfjson.Plan, resourceType string)`

Reply:
ah i see, so having Plan... in the function name is redundant because a plan resource is the first argument. makes sense

Reply:
Correct, the function is the verb, args are the nouns and adjectives

Reply:
Sometimes you need to disambiguate but having simple verb method names is typically a sign of good package design, i.e. that each package is focused. Because polymorphism in Go isn't a thing, if you do have several similar methods with slightly different input types, then you'll have to put the noun in the func name.